Color feature

Thorben0 opened the pull request:
# Colors
___
#### Diese Pull Request fügt bearbeitbare Farben zu den Notizen hinzu, die über einen Color Picker auswählbar sind.

Funktionen:
* Farbe für Notizen
* Farbe auswählbar über Color Picker
* Farbtemplates in Datenbank gespeichert
* Farben hinzufügen, löschen, bearbeiten
* Adaptive Font-Farbe für Notizen basierend auf Hintergrundfarbe
* Angezeigt auf Hauptseite und im Editor

[Link zur Website](https://memota.zernico.de)

---
### Gruppe 21:
* Thorben Edvard Stumpe
* Nico Franke
* Adrian Aljoscha Ruppert
* Benedikt Pröckl

Thorben0 committed:
Add color to notes

Thorben0 committed:
Update docs

viktorianer reviewed (approved):
Sehr gute Arbeit Thorben. Aber, leider ist das kein eigenes Feature. Ein Feature sollte ja immer MVC beinhalten. Das ist in diesen Fall sogar MVC, aber du erweiterst nur ein wenig ein anderes Feature. Teilpunkte kann ich problemlos geben. Für ein eigenes Feature erwarte ich aber von dir noch etwas mehr. 🥇

viktorianer commented on src/components/EditNoteDialog.vue:
```diff
@@ -1,13 +1,38 @@
+                  no-footer
+                  default-view="palette"
+                  :palette="[
+                    '#FFFFFF',
+                    '#ff9999',
+                    '#ffcc99',
+                    '#ffff99',
+                    '#99ff99',
+                    '#99ffff',
+                    '#99ccff',
+                    '#9999ff',
+                    '#cc99ff',
+                    '#ff99ff',
```
Wenn das aus DB kommen würde, oder mehr Logik hätte, würde ich das schon eher als eigenes Feature akzeptieren.

Thorben0 commented:
Danke, aber ich verstehe nicht, was ein leichtes Feature sonst ist. Es war ja vorher gar nicht möglich, die Farben der Notizen zu ändern. Die Farben werden hier pro Notiz auch in der Datenbank gespeichert. Bei den Beispielen für leichte Features stehen ja so Sachen wie dark mode oder header/footer, die gar kein MVC beinhalten.

viktorianer commented:
Die Beispiele sind alle MVC, bis auf dynamische Startseite. 

Du bekommst hier halbe Punktezahl, da es eine leichte Erweiterung von einem anderen Feature ist. Es ist nicht wirklich viel Aufwand.

Thorben0 committed:
Add color picker with custom templates

Thorben0 committed:
Finish color feature

- Update custom colors to note card
- Adapt font to note color
- Stylize new color picker

Thorben0 committed:
Update docs

daku-souru reviewed (approved):
💯

viktorianer reviewed (approved):
Das ist 💯 👍🏼
